### PR TITLE
walk the build graph in parallel

### PR DESCRIPTION
**Pull request by BrianHicks**
This makes `Coordinator` walk the build graph in parallel.

After some discussion on Zulip, I did this with async Rust instead of spawning threads directly. I think this could still use some improvement, but it works well for now. We can tune it with tokio-console or something soon!

### CODE REVIEW

**Comment by BrianHicks** [comment]
@roc-lang/rbt-reviewers I'm requesting review of this since it seems like a configuration failure in CI. The tests all work locally, and the lints all pass in CI. 👍

**Comment by bhansconnect** [comment]
You should be able to use `roc gen-stub-lib some-example-app.roc` on nightly now to work around the build issues here.

**Comment by BrianHicks** [comment]
amazing, thank you!

**Comment by BrianHicks** [comment]
@bhansconnect any idea what could be going on here? Looks like we're generating `libapp.so` correctly, but then `cargo test` complains it can't be found. Is it looking somewhere other than in the current directory, do you think? https://github.com/roc-lang/rbt/actions/runs/3374554604/jobs/5600303279

**Comment by BrianHicks** [comment]
ok, monkeyed around with LD_LIBRARY_PATH. Now it's saying this:

```
/home/runner/work/rbt/rbt/target/release/deps/host-e02e8b3c7e864cef: error while loading shared libraries: ./libapp.so: ELF file's phentsize not the expected size
```

File: src/coordinator.rs [hunk 1]
```diff
@@ -244,78 +266,157 @@ impl<'roc> Builder<'roc> {
...
-    pub fn run_next<R: Runner>(&mut self, runner: &R) -> Result<()> {
-        let id = match self.ready.pop() {
-            Some(id) => id,
-            None => anyhow::bail!("no work ready to do"),
-        };
+    /// Start any outstanding work according to our scheduling rules. Right
+    /// now that just means that we won't ever be running more jobs than
+    /// `self.max_local_jobs`.
+    async fn schedule(&mut self) -> Result<()> {
+        let maximum_schedulable = (self.max_local_jobs - self.running.len()).max(0);
```
Comment (rtfeldman) [inline src/coordinator.rs:324]:
```suggestion
        let maximum_schedulable = self.max_local_jobs.saturating_sub(self.running.len());
```

https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_sub

Comment (rtfeldman) [reply src/coordinator.rs:324]: Also possibly useful:

```rust
debug_assert!(self.running.len() <= self.max_local_jobs);
```

**Review by rtfeldman — APPROVED** [review]
Looks lovely! 🎉

**Pull request merged**